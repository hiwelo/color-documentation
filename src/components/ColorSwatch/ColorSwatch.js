import React from 'react';
import PropTypes from 'prop-types';
import defaultOptions from '../../config/defaultOptions';

const ColorSwatch = props => <></>;

ColorSwatch.propTypes = {
  /**
   * Specify the color used by this color swatch
   */
  color: PropTypes.string.isRequired,

  /**
   * Specify the options for this component
   */
  options: PropTypes.object,
};

ColorSwatch.defaultProps = {
  options: defaultOptions,
};

export default ColorSwatch;