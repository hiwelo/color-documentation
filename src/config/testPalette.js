const testPalette = {
  generic: {
    black: "#000",
    white: "#fff"
  }
};

export default testPalette;
