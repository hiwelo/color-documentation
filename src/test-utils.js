import { render } from '@testing-library/react';

/**
 * This file is compiling all useful methods
 * and consts used by unit tests.
 */

const queryByTestId = (testId) => {
  return document.querySelector(`[data-testid="${testId}"]`);
};

const queryParentByTestId = (testId) => {
  return queryByTestId(testId).parentElement;
};

const getTag = (element) => {
  return element.tagName.toLowerCase();
}

const doSnapshotTest = (DOMNode) => {
  const { container } = render(DOMNode);

  expect(container.firstChild).toMatchSnapshot();
};

export { queryByTestId, queryParentByTestId, getTag, doSnapshotTest };
