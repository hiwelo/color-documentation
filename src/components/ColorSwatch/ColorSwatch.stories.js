import React from 'react';
import { storiesOf } from '@storybook/react';
import ColorSwatch from './ColorSwatch';
import testPalette from '../../config/testPalette';

storiesOf('ColorSwatch', module)
  .add('overview', () => (
    <ColorSwatch color={testPalette.generic.black} />
  ))
;
