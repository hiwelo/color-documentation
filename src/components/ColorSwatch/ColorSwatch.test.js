import React from "react";
import { render } from "@testing-library/react";
import ColorSwatch from "./ColorSwatch";
import { doSnapshotTest } from "../../test-utils.js";
import testPalette from "../../config/testPalette.js";

const palette = testPalette[Object.keys(testPalette)[0]];

const defaultProps = {
  color: palette[Object.keys(palette)[0]]
};

describe("ColorSwatch", () => {
  it("should render without crashing", () => {
    render(<ColorSwatch {...defaultProps} />);
  });

  it("should match snapshot", () => {
    doSnapshotTest(<ColorSwatch {...defaultProps} />);
  });
});
