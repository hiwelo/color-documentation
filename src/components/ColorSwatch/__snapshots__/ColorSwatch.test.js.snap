// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSwatch should match snapshot 1`] = `null`;
